controller/users.js

```diff
@@ -0,0 +1,84 @@
+const { User } = require("./../service/schemas/user.js");
+const service = require("./../service/users");
+const jwt = require("jsonwebtoken");
+require("dotenv").config();
+const secret = process.env.JWT_SECRET;
+
+const getAll = async (req, res, next) => {
+  try {
+    const results = await User.find();
+    res.status(200).json(results);
+  } catch (err) {
+    console.error(err.message);
+    next(err);
+  }
+};
+
+const register = async (req, res, next) => {
+  const { email, password } = req.body;
+  const user = await service.getUser({ email });
+
+  if (user) return res.status(409).json({ message: "Email in use" });
+
+  try {
+    const newUser = new User({ email });
+    newUser.setPassword(password);
+    await newUser.save();
+    res.status(201).json({
+      user: {
+        email,
+        subscription: "starter",
+      },
+    });
+  } catch (err) {
+    next(err);
+  }
+};
+
+const login = async (req, res, next) => {
+  try {
+    const { email, password } = req.body;
+    const user = await service.getUser({ email });
+
+    if (!user || !user.validPassword(password))
+      return res.status(401).json({ message: "Email or password is wrong" });
+
+    const payload = {
+      id: user.id,
+      email: user.email,
+    };
+
+    const token = jwt.sign(payload, secret, { expiresIn: "1h" });
+    await service.updateUser(user.id, { token });
+    res
+      .status(200)
+      .json({ token, user: { email, subscription: user.subscription } });
+  } catch (err) {
+    next(err);
+  }
+};
+
+const logout = async (req, res, next) => {
+  try {
+    const user = await service.getUser({ _id: req.user._id });
+    if (!user) return res.status(401).json({ message: "Not authorized" });
+    await service.updateUser(user.id, { token: null });
+    res.sendStatus(204);
+  } catch (err) {
+    next(err);
+  }
+};
+
+const current = async (req, res, next) => {
+  try {
+    const user = await service.getUser({ token: req.user.token });
+    if (!user) return res.status(401).json({ message: "Not authorized" });
+    res
+      .status(200)
+      .json({ email: user.email, subscription: user.subscription });
```

Comment:
zgaduje, ze prettier pewnie sam Ci rozbil to na 3 linijki. Wobec czego pewnie ladniej byloby:

```js
const {email, subscription} = user;
res.status(200).json({email,subscription});
```